### PR TITLE
Adapted file for use in editor.swagger.io

### PR DESCRIPTION
**Pull request by ChristianMaehler**
I didn't succeed to import the JSON file in editor.swagger.io. I solved the problem like this:
- import the yml file into editor.swagger.io ==> success
- export it without modification as JSON in file2.json
- import the file2.json in editor.swagger.io ==> success
- File can now be used https://editor.swagger.io

The pull request comprises the file2.json content without further modifications. Hope that it is useful for you.

### CODE REVIEW

File: specs/maas-v1.json [hunk 1]
```diff
@@ -677,49 +641,39 @@
       "maximum": 180
     },
     "legGeometry": {
-      "x-oad-type": "object",
       "type": "object",
-      "title": "Leg geometry",
```
Comment (brylie) [inline specs/maas-v1.json]: Why do we remove the title here?

File: specs/maas-v1.json [hunk 2]
```diff
@@ -634,25 +605,18 @@
...
         "stopId": {
-          "x-oad-type": "string",
-          "type": "string",
-          "title": "Stop ID"
+          "type": "string"
         },
         "stopCode": {
-          "x-oad-type": "string",
-          "type": "string",
-          "title": "Stop code"
```
Comment (brylie) [inline specs/maas-v1.json]: Why do we remove the title here?

File: specs/maas-v1.json [hunk 3]
```diff
@@ -677,49 +641,39 @@
...
-          "title": "Points"
+          "minLength": 1
         }
       }
     },
     "time": {
-      "x-oad-type": "integer",
       "description": "A UTC timestamp (number of milliseconds in a Date object since January 1, 1970, 00:00:00)",
       "type": "integer",
-      "title": "Time",
```
Comment (brylie) [inline specs/maas-v1.json]: Why do we remove the title here?

File: specs/maas-v1.json [hunk 4]
```diff
@@ -677,49 +641,39 @@
...
-      "format": "int32"
+      "maximum": 2147483647,
+      "minimum": 0
     },
     "duration": {
-      "x-oad-type": "integer",
-      "type": "integer",
-      "title": "Duration",
```
Comment (brylie) [inline specs/maas-v1.json]: Why do we remove the title here?

File: specs/maas-v1.json [hunk 5]
```diff
@@ -677,49 +641,39 @@
...
       "description": "A duration of some time (relative to time) in milliseconds",
-      "format": "int32"
+      "type": "integer",
+      "maximum": 2147483647,
+      "minimum": 0
     },
     "distance": {
-      "x-oad-type": "integer",
-      "type": "integer",
-      "title": "Distance",
```
Comment (brylie) [inline specs/maas-v1.json]: Why do we remove the title here?

File: specs/maas-v1.json [hunk 6]
```diff
@@ -677,49 +641,39 @@
...
       "description": "The estimated distance travelled in the leg (in meters)",
-      "format": "int32"
+      "type": "integer",
+      "minimum": 0
     },
     "fare": {
-      "x-oad-type": "object",
-      "type": "object",
-      "title": "Fare",
```
Comment (brylie) [inline specs/maas-v1.json]: Why do we remove the title here?

File: specs/maas-v1.json [hunk 7]
```diff
@@ -677,49 +641,39 @@
...
-      "description": "Arbitrary fare data that MaaS will use internally"
+      "description": "Arbitrary fare data that MaaS will use internally",
+      "type": "object"
     },
     "mode": {
-      "x-oad-type": "string",
-      "type": "string",
-      "title": "Mode",
```
Comment (brylie) [inline specs/maas-v1.json]: Why do we remove the title here?

File: specs/maas-v1.json [hunk 8]
```diff
@@ -458,42 +434,54 @@
...
-          "type": "string",
-          "title": "Last name",
-          "description": "Last name of the customer (e.g. Doe)"
+          "description": "Last name of the customer (e.g. Doe)",
+          "type": "string"
         },
         "phone": {
-          "x-oad-type": "string",
-          "type": "string",
-          "title": "Phone",
```
Comment (brylie) [inline specs/maas-v1.json]: Why do we remove the title here?

File: specs/maas-v1.json [hunk 9]
```diff
@@ -1,4 +1,23 @@
 {
+  "swagger": "2.0",
+  "info": {
+    "version": "v1",
+    "title": "MaaS-TSP",
+    "description": "This is a API specification of REST endpoints that a Transport Service Provider (TSP) should implement to receive messages from MaaS. It is written in machine readable [Swagger](http://swagger.io/) format, so thatAPI endpoints, validators and test clients can be generated from the documentation.\n\nIn addition to this documentation, MaaS maintains a [reference implementation (**TODO**)](https://github.com/maasglobal/maas-tsp-api) with test cases. It is also running as a [reference service (**TODO**)](https://tsp.maas.global/).\n\nRight now MaaS TSP API consists of the Booking API. In addition, a need for customer feedback API (for mediating customer feedback to a TSP) and administration API (for changing API keys etc.) may be defined.\n",
+    "termsOfService": "http://api.maas.global/terms/",
+    "contact": {
+      "name": "MaaS API Team",
+      "email": "developers@maas.fi",
```
Comment (brylie) [inline specs/maas-v1.json]: This should probably be `tech@maas.global`

**Comment by brylie** [comment]
For what it's worth, the title fields in the schema were to improve the appearance of the documentation when rendered in Swagger UI. Please leave the title fields in place.

**Comment by ChristianMaehler** [comment]
Oh, I didn't remove the titles manually - I didn't change anything :-) I only imported your yml and exported it again as json . It was done automatically in the editor.swagger.io ... as far as I can see in the diff, the parameters are sometimes changed in location in the JSON; I didn't check whether there were really removed parameters.

**Comment by brylie** [comment]
OK, just double check. If they are not removed, we can merge. Otherwise, just add them back.

**Comment by brylie** [comment]
I think the problem is that the YML and JSON files are out of sync. The JSON file had 'title' attributes for many of the schema items, while the YML file doesn't. Please consider re-adding the title attributes, so we don't lose those.

**Comment by brylie** [comment]
Upon reflection, I realize that we should probably only maintain one copy of the specification in a single format. Since YAML is easy to read, and supported by Swagger Editor, I offer that we should actually remove the JSON version from this repository.

What are your thoughts @ChristianMaehler?

**Comment by ChristianMaehler** [comment]
@brylie Although I'm a big fan of JSON :-) , I find it hard to read for an OpenAPI specification; YAML seems to be much more readable. In addition, it will be easier to maintain one version - so, I fully agree. I don't know whether there are some pitfalls using YML only, but atm I can't see any. So my vote is: yes, remove the JSON and keep the YAML.

**Comment by brylie** [comment]
@ChristianMaehler thanks for your input. Would you like to remove the JSON as part of this PR, so you get credit for your contribution?

If you remove the JSON and push the change, I will review/merge this PR :-)

**Comment by brylie** [comment]
Closing this PR, as I have removed the JSON spec from the repo.

**Pull request closed without merging**